### ISSUE TITLE
Build failure under Bazel on debian

### ISSUE THREAD

**Issue by alexeagle** [google/brotli#650]
I'm on gLinux (google-internal)

`WORKSPACE`
```
http_archive(
    name = "org_brotli",
    url = "https://github.com/google/brotli/archive/v1.0.2.zip",
    strip_prefix = "brotli-1.0.2",
    sha256 = "b43d5d6bc40f2fa6c785b738d86c6bbe022732fe25196ebbe43b9653a025920d",
)
```

```

$ bazel info release
release 0.11.1
$ cat /etc/issue
Debian GNU/Linux buster/sid
$ bazel build @org_brotli//:brotlidec
INFO: Analysed target @org_brotli//:brotlidec (0 packages loaded).
INFO: Found 1 target...
ERROR: /usr/local/google/home/alexeagle/.cache/bazel/_bazel_alexeagle/5c094f47f582dc47fc598194e749d5c0/external/org_brotli/BUILD:140:1: C++ compilation of rule '@org_brotli//:brotlidec' failed (Exit 1)
external/org_brotli/c/dec/decode.c: In function 'DecodeVarLenUint8':
external/org_brotli/c/dec/decode.c:154:10: error: this statement may fall through [-Werror=implicit-fallthrough=]
       if (bits == 0) {
          ^
external/org_brotli/c/dec/decode.c:160:5: note: here
     case BROTLI_STATE_DECODE_UINT8_SHORT:
     ^~~~
external/org_brotli/c/dec/decode.c:171:14: error: this statement may fall through [-Werror=implicit-fallthrough=]
       *value = bits;
       ~~~~~~~^~~~~~
external/org_brotli/c/dec/decode.c:174:5: note: here
     case BROTLI_STATE_DECODE_UINT8_LONG:
     ^~~~
external/org_brotli/c/dec/decode.c: In function 'DecodeMetaBlockLength':
external/org_brotli/c/dec/decode.c:208:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header = BROTLI_STATE_METABLOCK_HEADER_EMPTY;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:211:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_EMPTY:
       ^~~~
external/org_brotli/c/dec/decode.c:219:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header = BROTLI_STATE_METABLOCK_HEADER_NIBBLES;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:222:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_NIBBLES:
       ^~~~
external/org_brotli/c/dec/decode.c:233:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header = BROTLI_STATE_METABLOCK_HEADER_SIZE;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:236:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_SIZE:
       ^~~~
external/org_brotli/c/dec/decode.c:248:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header =
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
             BROTLI_STATE_METABLOCK_HEADER_UNCOMPRESSED;
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:252:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_UNCOMPRESSED:
       ^~~~
external/org_brotli/c/dec/decode.c:270:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header = BROTLI_STATE_METABLOCK_HEADER_BYTES;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:273:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_BYTES:
       ^~~~
external/org_brotli/c/dec/decode.c:282:38: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_metablock_header = BROTLI_STATE_METABLOCK_HEADER_METADATA;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:285:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER_METADATA:
       ^~~~
external/org_brotli/c/dec/decode.c: In function 'ReadHuffmanCode':
external/org_brotli/c/dec/decode.c:718:12: error: this statement may fall through [-Werror=implicit-fallthrough=]
         if (s->sub_loop_counter != 1) {
            ^
external/org_brotli/c/dec/decode.c:730:7: note: here
       case BROTLI_STATE_HUFFMAN_SIMPLE_SIZE:
       ^~~~
external/org_brotli/c/dec/decode.c:736:29: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->sub_loop_counter = 0;
         ~~~~~~~~~~~~~~~~~~~~^~~
external/org_brotli/c/dec/decode.c:738:7: note: here
       case BROTLI_STATE_HUFFMAN_SIMPLE_READ: {
       ^~~~
external/org_brotli/c/dec/decode.c:738:46: error: this statement may fall through [-Werror=implicit-fallthrough=]
       case BROTLI_STATE_HUFFMAN_SIMPLE_READ: {
                                              ^
external/org_brotli/c/dec/decode.c:746:7: note: here
       case BROTLI_STATE_HUFFMAN_SIMPLE_BUILD: {
       ^~~~
external/org_brotli/c/dec/decode.c:787:29: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_huffman = BROTLI_STATE_HUFFMAN_LENGTH_SYMBOLS;
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:790:7: note: here
       case BROTLI_STATE_HUFFMAN_LENGTH_SYMBOLS: {
       ^~~~
external/org_brotli/c/dec/decode.c: In function 'DecodeContextMap':
external/org_brotli/c/dec/decode.c:964:31: error: this statement may fall through [-Werror=implicit-fallthrough=]
       s->substate_context_map = BROTLI_STATE_CONTEXT_MAP_READ_PREFIX;
       ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:966:5: note: here
     case BROTLI_STATE_CONTEXT_MAP_READ_PREFIX: {
     ^~~~
external/org_brotli/c/dec/decode.c:966:48: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case BROTLI_STATE_CONTEXT_MAP_READ_PREFIX: {
                                                ^
external/org_brotli/c/dec/decode.c:984:5: note: here
     case BROTLI_STATE_CONTEXT_MAP_HUFFMAN:
     ^~~~
external/org_brotli/c/dec/decode.c:989:31: error: this statement may fall through [-Werror=implicit-fallthrough=]
       s->substate_context_map = BROTLI_STATE_CONTEXT_MAP_DECODE;
       ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:991:5: note: here
     case BROTLI_STATE_CONTEXT_MAP_DECODE: {
     ^~~~
external/org_brotli/c/dec/decode.c:991:43: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case BROTLI_STATE_CONTEXT_MAP_DECODE: {
                                           ^
external/org_brotli/c/dec/decode.c:1039:5: note: here
     case BROTLI_STATE_CONTEXT_MAP_TRANSFORM: {
     ^~~~
external/org_brotli/c/dec/decode.c: In function 'CopyUncompressedBlockToOutput':
external/org_brotli/c/dec/decode.c:1322:34: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->substate_uncompressed = BROTLI_STATE_UNCOMPRESSED_WRITE;
         ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:1325:7: note: here
       case BROTLI_STATE_UNCOMPRESSED_WRITE: {
       ^~~~
external/org_brotli/c/dec/decode.c: In function 'BrotliDecoderDecompressStream':
external/org_brotli/c/dec/decode.c:2034:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_METABLOCK_BEGIN;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2036:7: note: here
       case BROTLI_STATE_METABLOCK_BEGIN:
       ^~~~
external/org_brotli/c/dec/decode.c:2039:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_METABLOCK_HEADER;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2041:7: note: here
       case BROTLI_STATE_METABLOCK_HEADER:
       ^~~~
external/org_brotli/c/dec/decode.c:2110:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_HUFFMAN_CODE_1;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2112:7: note: here
       case BROTLI_STATE_HUFFMAN_CODE_1: {
       ^~~~
external/org_brotli/c/dec/decode.c:2117:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_HUFFMAN_CODE_2;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2120:7: note: here
       case BROTLI_STATE_HUFFMAN_CODE_2: {
       ^~~~
external/org_brotli/c/dec/decode.c:2125:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_HUFFMAN_CODE_3;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2128:7: note: here
       case BROTLI_STATE_HUFFMAN_CODE_3: {
       ^~~~
external/org_brotli/c/dec/decode.c:2140:45: error: this statement may fall through [-Werror=implicit-fallthrough=]
       case BROTLI_STATE_METABLOCK_HEADER_2: {
                                             ^
external/org_brotli/c/dec/decode.c:2163:7: note: here
       case BROTLI_STATE_CONTEXT_MODES:
       ^~~~
external/org_brotli/c/dec/decode.c:2168:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_CONTEXT_MAP_1;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2170:7: note: here
       case BROTLI_STATE_CONTEXT_MAP_1:
       ^~~~
external/org_brotli/c/dec/decode.c:2178:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_CONTEXT_MAP_2;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2180:7: note: here
       case BROTLI_STATE_CONTEXT_MAP_2:
       ^~~~
external/org_brotli/c/dec/decode.c:2206:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_TREE_GROUP;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2208:7: note: here
       case BROTLI_STATE_TREE_GROUP:
       ^~~~
external/org_brotli/c/dec/decode.c:2302:18: error: this statement may fall through [-Werror=implicit-fallthrough=]
         s->state = BROTLI_STATE_DONE;
         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~
external/org_brotli/c/dec/decode.c:2304:7: note: here
       case BROTLI_STATE_DONE:
       ^~~~
cc1: all warnings being treated as errors
Target @org_brotli//:brotlidec failed to build
```

**Comment by eustas** [comment]
Going to look at this soon. Most likely, we will have to explicitly disable this warning.

**Comment by alexeagle** [comment]
Thanks, right now we're not able to run Angular's tests on our workstations since we have a test for the size of our brotli-bundled output.

**Comment by eustas** [comment]
Going to push the fix to github in few hours. Sorry for the delay.

**Comment by eustas** [comment]
Should be fixed with #651

**Comment by alexeagle** [comment]
Thanks!!
